### PR TITLE
fix an unnecessary downcast

### PR DESCRIPTION
**Pull request by jsteemann**

### CODE REVIEW

File: lib/Basics/files.cpp [hunk 1]
```diff
@@ -1671,7 +1671,7 @@ static bool CopyFileContents(int srcFD, int dstFD, ssize_t fileSize,
         break;
       }
 
-      if ((TRI_read_t)TRI_WRITE(dstFD, buf, nRead) != nRead) {
+      if (static_cast<int64_t>(TRI_WRITE(dstFD, buf, static_cast<TRI_write_t>(nRead))) != static_cast<int64_t>(nRead)) {
```
Comment (neunhoef) [inline lib/Basics/files.cpp]: This is actually a comment about line 1668: TRI_READ returns a signed value (ssize_t) and < 0 has a special meeing. Assigning this  to nRead which is unsigned is not good, in particular if we then compare (nRead < 1), so we miss errors! I fear we need to distinguish return types and length arguments. The usage of int64_t here might be a way out, but let's fix this rather carefully!

Comment (jsteemann) [reply lib/Basics/files.cpp]: will fix it

Comment (jsteemann) [reply lib/Basics/files.cpp]: Fixed it and added a few tests for it.

Comment (jsteemann) [reply lib/Basics/files.cpp]: Also created a follow-up issue, as we do not seem to use the optimized file copy method based on splice for Linux anymore: https://github.com/arangodb/planning/issues/1310

**Comment by jsteemann** [comment]
Any better now?

**Pull request merged**